File: 1. 两数之和.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> record = new HashMap<Integer, Integer>();
        int[] ans = new int[2];
        for (int i = 0; i < nums.length; i++) {
            record.put(nums[i], i);
        }
        for (int i = 0; i < nums.length - 1; i++) {
            if (record.containsKey(target - nums[i])) {
                if (record.get(target - nums[i]) == i) {
                    continue;
                }
                ans[0] = i;
                ans[1] = record.get(target - nums[i]);
                break;
            }
        }
        return ans;
    }
}

File: LeetCode_python/1两数之和.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}
        for i, x in enumerate(nums):
            if (target - x) in hash_map:
                return [i, hash_map[target - x]]
            hash_map[x] = i
        return []